src/LanguageFunctions.ts

```diff
@@ -134,33 +134,43 @@ export async function RefreshXlfFilesFromGXlf(sortOnly?: boolean, matchXlfFileUr
     NumberOfUpdatedSources: number;
     NumberOfRemovedTransUnits: number;
 }> {
+    sortOnly = (sortOnly === null) ? false : sortOnly; //TODO: Test this!
+    const useMatchingSetting: boolean = (Settings.GetConfigSettings()[Setting.MatchTranslation] === true); 
+    let currentUri: vscode.Uri | undefined = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.uri : undefined;
+    let gXlfFileUri = (await WorkspaceFunctions.GetGXlfFile(currentUri));
+    let langFiles = (await WorkspaceFunctions.GetLangXlfFiles(currentUri));
+    let useExternalTranslationTool: boolean = Settings.GetConfigSettings()[Setting.UseExternalTranslationTool];
+    return (await __RefreshXlfFilesFromGXlf(gXlfFileUri, langFiles, useExternalTranslationTool, useMatchingSetting, sortOnly, matchXlfFileUri));
+    
+}
+
+export async function __RefreshXlfFilesFromGXlf(gXlfFilePath: vscode.Uri, langFiles: vscode.Uri[], useExternalTranslationTool: boolean, useMatchingSetting?: boolean, sortOnly?: boolean, matchXlfFileUri?: vscode.Uri, printToConsole: boolean = false): Promise<{
```

Comment:
when will printToConsole ever be true?

Reply:
maybe it's better to skip this parameter and in console_log somehow check if debugger is running?

Reply:
> when will printToConsole ever be true?

Hopefully never? ;) Is the console.log by design or just left in for easy debugging?


> maybe it's better to skip this parameter and in console_log somehow check if debugger is running?

Do you know any way? I tried vscode.debug.activeDebugSession but it returns `undefined`. I think it's due to the tests beeing run in a new window and I'm trying to figure out if it's dependent on a workspace. I see that Waldo is using settings that could be the way to go so I'll look into that.

Reply:
A setting would do it, perfect for troubleshooting.

I realized that we also have src\Logging.ts that could be used for logging. Maybe use that (combined with setting) instead? Logging is (at least partly) used in src\PowerShell.ts... seems to never have been completed :D

Reply:
So from what I've found there's no problem with loading any json file as a config file. However  since the settings are loaded and reloaded every now and then I can't see how I could reach a simple and good solution without it being a janky side loader or a huge rework.

I was just about to give up when I thought that I'd try to use my brain one last time. And it worked. Its so simple I hurt my own feelings. In launch.json just add the path to a workspace file as the first item in `args[]` and that workspace will be opened. Works with both configurations.

```json
launch.json
       {
            "name": "Extension Tests",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": [
                "/path-to-my-app/MY_App.code-workspace",*
                "--extensionDevelopmentPath=${workspaceFolder}",
                "--extensionTestsPath=${workspaceFolder}/out/test"
            ],
            "outFiles": [
                "${workspaceFolder}/out/test/**/*.js"
            ],
            "preLaunchTask": "npm: watch"
        }
```

Reply:
a9410ab  Added empty TestApp with workspace add configured it in launch.json. Works like a charm :)